app/main/views/suppliers.py

```diff
@@ -517,80 +516,48 @@ def submit_duns_number():
         ), 400
 
 
-@main.route('/company-name', methods=['GET'])
-def company_name():
-    form = CompanyNameForm()
+@main.route('/company-details', methods=['GET', 'POST'])
+def company_details():
+    form = CompanyPublicContactInformationForm()
+
+    if request.method == "POST":
+        if form.validate_on_submit():
+            session[form.company_name.name] = form.company_name.data
+            session[form.email_address.name] = form.email_address.data
+            session[form.phone_number.name] = form.phone_number.data
+            session[form.contact_name.name] = form.contact_name.data
+            return redirect(url_for(".create_your_account"))
+        else:
+            current_app.logger.warning(
+                "suppliercreate.fail: duns:{duns} company_name:{company_name} {form_errors}",
+                extra={
+                    'duns': session.get('duns_number'),
+                    'company_name': session.get('company_name'),
```

Comment:
Actually, I think maybe this should not have `company_name` and only DUNS number, as that's the only thing that will have been set in the session up to now?

Company name, etc. only get set if the form validates.  If it hasn't validated then none of these things will have values in the session.

We could use the values from the form rather than the values from the session to populate this?  But to be honest I'm not sure there's much value in logging validation errors at all.

Reply:
I think all of the other instances of this logging grab the data from the form, so getting it from the session at all is kinda weird and probably wrong.

Although as you say, it's questionable the value it has...